packages/layer-composer/src/generators/vessel-events/vessel-events.ts

```diff
@@ -68,39 +51,66 @@ class VesselsEventsGenerator {
       })
     }
 
-    const source: VesselsEventsSource = {
-      id,
+    const pointsSource: VesselsEventsSource = {
+      id: `${id}_points`,
       type: 'geojson',
       data: newData,
     }
-    return [source]
+
+    if (!config.showTrackSegments) {
+      return [pointsSource]
+    }
+    // TODO review performance memoization
+
+    const segments = memoizeCache[config.id].getVesselSegmentsGeojson(
+      track,
+      data,
+      start,
+      end,
+      currentEventId
```

Comment:
Another potential performance fix on timebar hover: segments GeoJSON could be first generated entirely. Then another pass to attach the `active` property to geojson features. This way `getVesselSegmentsGeojson` can be memoized without depending on `currentEventId`.

Reply:
It might even be better to do:
- do not attach `active` property
- modify expression here : https://github.com/GlobalFishingWatch/frontend/pull/368/files#diff-3a95f5df66ac12142259807dff47e753fd374a82483431c24ebb6dfbcef048fdR86
  to directly compare `event.id` and `currentEventId`

Reply:
👏 it works way better! see: https://github.com/GlobalFishingWatch/frontend/pull/368/commits/e22723264fdc42d16d3603fd6421d2caf774a1d2